doc/api/url.md

```diff
@@ -1562,16 +1565,9 @@ A `URIError` is thrown if the `auth` property is present but cannot be decoded.
 
 `url.parse()` uses a lenient, non-standard algorithm for parsing URL
 strings. It is prone to security issues such as [host name spoofing][]
-and incorrect handling of usernames and passwords.
-
-`url.parse()` is an exception to most of the legacy APIs. Despite its security
-concerns, it is legacy and not deprecated because it is:
-
-* Faster than the alternative WHATWG `URL` parser.
-* Easier to use with regards to relative URLs than the alternative WHATWG `URL` API.
```

Comment:
Do we have a solution for relative URLs using WHATWG?

Reply:
If you know the base URL, then yes. If not, you have to fake it, perhaps with an identifiably bogus base URL. In that second case, it's a solution, but perhaps not a particularly elegant one.

My opinion only, but a lot of the current problems with `url.parse()` result from the design decision (probably sensible at the time) to handle both full URLs and relative URLs without knowing the base URL.

Ref: https://github.com/nodejs/node/issues/12682
Ref: https://github.com/whatwg/url/issues/531

Reply:
Thanks @Trott 

My worry is that marking `url.parse()` as deprecated is that we introduce new problems when someone assumes `new URL()` will work as a replacement but it wont for relative URLS. And in fact, I think most code I've seen is intending to use it with relative urls like that of `url.parse(req.url)`.

I remember when this was "deprecated" the first time, it confused a lot of people and then it was reverted to say "legacy" which I think was the right decision.

Or maybe the solution here is to document how to parse relative urls, such as `req.url`.

Reply:
There's https://github.com/whatwg/url/issues/421 that would fit the use-case, unfortunately the proposal look rather stalled.

Reply:
I’ve seen that one too. 

It has a label “needs implementer interest”. Does that need to be a browser or can we say Node.js is interested in implementing?

Reply:
The proposal comes from Node.js, I think it's implied that Node.js is interested, I think we need other implementers as well.

Reply:
@aduh95 The discussion around https://github.com/whatwg/url/issues/421 more or less has been continued in https://github.com/whatwg/url/issues/531.

Reply:
@styfle I have spent a lot of effort to find a solution for relative URLs that agrees with the WHATWG Standard, see e.g. [this comment] and also this [URL Specification] and this [library]. 

This was indeed far from trivial due to design decisions made in the WHATWG standard. 

I have an idea for a simple API ready, but I stopped working on it for a while. In part because it is unlikely that the WHATWG standard can incrementally add support for relative URLs. This leads to an understandable resistance, which nonetheless is frustrating to me and something that I don’t yet know how to solve.

The other reason is simply being preoccupied with other work at the moment. 

Suggestions and ideas are appreciated!

[this comment]: https://github.com/nodejs/node/issues/12682#issuecomment-834828940
[library]: https://github.com/alwinb/spec-url
[URL Specification]: https://github.com/alwinb/url-specification

Reply:
Looks interesting. I think WHATWG is probably the wrong place to standardize this since browsers aren't interested. I think [WinterCG](https://wintercg.org/) is a better choice for standardization of relative url parsing since it is focused on server-side JS.

Reply:
Personally I think that marking this API as deprecated should've been gated on a suitable replacement being made available, and perhaps that would've introduced some incentive to come to a consensus on what that replacement should look like. 

Relative URLs are a thing, and we need a suitable API to replace this one.